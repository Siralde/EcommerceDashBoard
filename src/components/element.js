import React, {useState} from 'react';
import Image from "../assets/images/conga1.png";
import Edit from "../assets/images/edit.png";
import Trash from "../assets/images/trash.png";
import Plus from "../assets/images/add.svg";
import Cancel from "../assets/images/cancel.png";
import Check from "../assets/images/check.png";

export function UpdateElement({id, header, primaryDescription, secundaryDescription, img, updateElement, url, setUpdate, products}) 
{
    
    const [allValues, setAllValues] = useState({
        id,
        full_name: header,
        email: primaryDescription,
        telephone: secundaryDescription,
        name: header,
        description: primaryDescription,
        price: secundaryDescription,
        img
    });

    const changeHandler = e => {
        setAllValues({...allValues, [e.target.name]: e.target.value})
    }

    return (
        <div className="crud_element">
            <form onSubmit={(event) => updateElement(event, url, id, allValues, setUpdate)}>
                <img src={Image} alt="" />
                <div className="crud_element_body">
                    <div>
                        <input 
                            name={products ? "name" : "full_name"}
                            defaultValue={header}
                            onChange={changeHandler}
                        /> 
                        <img src={Image} alt="@some_lego_guy" />
                    </div>
                    <div>
                        <input 
                            name={products ? "description" : "email"}
                            defaultValue={primaryDescription}
                            onChange={changeHandler}
                        />
                    </div>
                    <div>
                        <input 
                            name={products ? "price": "telephone" }
                            defaultValue={secundaryDescription}
                            onChange={changeHandler}
                        />
                    </div>
                    <div className="crud_element_interactions">    
                        <button type="submit"><img src={Check} alt="Check" /></button>
                        <img src={Cancel} alt="Cancel" onClick={() => setUpdate(false)}/>
                    </div>
                </div>
            </form>
        </div>
    )
};


export function CreateElement( {updateElement, products}) 
{
    const [update, setUpdate] = useState(false)

    return (
        <div>
            {update 
                ?
                <UpdateElement updateElement={updateElement} setUpdate={setUpdate} products={products}/>
                :
                (<div className="new_element">
                    <img src={Plus} alt="Plus" onClick={ () => { setUpdate(!update) } } />
                </div>
                )
            }
        </div>
    )
};

export function ReadElement({id, header, primaryDescription, secundaryDescription, img, updateElement, deleteElement, url, products}) 
{

    const [update, setUpdate] = useState(false)

    return (
        <div>
            {update 
                ?
                <UpdateElement
                    id={id} 
                    header={header} 
                    primaryDescription={primaryDescription}
                    secundaryDescription={secundaryDescription}
                    img={id}
                    updateElement={updateElement}
                    url={url}
                    setUpdate={setUpdate}
                    products={products}
                />  
                :
                (<div className="crud_element">
                    <img src={Image} alt="" />
                    <div className="crud_element_body">
                        <div>
                            <p>{header}</p> 
                            <img src={Image} alt="@some_lego_guy" />
                        </div>

                        <p>{primaryDescription} </p>
                        <p>{secundaryDescription} </p>

                        <div className="crud_element_interactions">  
                        
                            <img src={Edit} alt="Edit" onClick={() => setUpdate(!update)}/>
                            <img src={Trash} alt="Trash" onClick={() => deleteElement(url, id)}/>
                        </div>
                    </div>
                </div>)
            }
        </div>
    )
};