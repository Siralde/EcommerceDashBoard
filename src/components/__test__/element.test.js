import React from 'react';
import ReactDOM from 'react-dom';
import { CreateElement } from '../element';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CreateElement/>, div);
})

it("renders without Create Element correctly", () => {
    const {getByTestId} = render(<CreateElement setUpdate={() => {}}/>)
    expect(getByTestId("createElement")).toHaveClass("new_element");
})

it("matches snapshot", () => {
    const tree = renderer.create(<CreateElement/>).toJSON();
    expect(tree).toMatchSnapshot();
})