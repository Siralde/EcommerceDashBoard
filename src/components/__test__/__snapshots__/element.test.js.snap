// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="new_element"
  data-testid="createElement"
>
  <img
    alt="Plus"
    onClick={[Function]}
    src="add.svg"
  />
</div>
`;
